db.go

```diff
@@ -1464,7 +1464,7 @@ func (db *DB) CalcRestoreTarget(ctx context.Context, opt RestoreOptions) (*Repli
 		}
 
 		// Use the latest replica if we have multiple candidates.
-		if !updatedAt.After(target.updatedAt) {
+		if generation == target.generation && !updatedAt.After(target.updatedAt) {
```

Comment:
Why was this change needed?

Reply:
I had an issue when restoring a database with a single generation and a target.updateAt set at 0, then the condition was true and at the end there was no replica available.

Reply:
Even though the PR itself was rejected I'm interested in understanding this issue if it's reproducible with the S3 gateway of Sorj?

Reply:
Looking into the code I believe the problem was that your replica implementation returned a generation without updatedAt set to the latest snapshot/WAL file which would've been wrong as otherwise it would always be >0.